File: from-1201-to-1300/1267-count-servers-that-communicate/solution.go
func countServers(grid [][]int) int {
    m, n := len(grid), len(grid[0])
    rows, cols := make([]int, m), make([]int, n)
    servers := 0
    for i := 0; i < m; i ++ {
        for j := 0; j < n; j ++ {
            if grid[i][j] == 1 {
                servers += 1
                rows[i] += 1
                cols[j] += 1
            }
        }
    }

    for i := 0; i < m; i ++ {
        for j := 0; j < n; j ++ {
            if grid[i][j] == 1 && rows[i] == 1 && cols[j] == 1 {
                servers -= 1
            }
        }
    }

    return servers
}

File: from-1201-to-1300/1267-count-servers-that-communicate/solution.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])
        rows, cols = [0] * m, [0] * n
        
        servers = 0
        for r in range(m):
            for c in range(n):
                if grid[r][c]:
                    servers += 1
                    rows[r] += 1
                    cols[c] += 1

        for r in range(m):
            for c in range(n):
                if grid[r][c] and rows[r] == 1 and cols[c] == 1:
                    servers -= 1

        return servers